### PR TITLE
[SFN] Support for Singleton Literal Array Access Unpacking

### PR DESCRIPTION
**Pull request by MEPalma**
## Motivation
AWS StepFunctions automatically unpacks singleton JsonPath results obtained through singleton literal array access, such as `$.items[0]`. These changes add support for such behaviour. Addresses: #11321

## Changes
- Support for singleton literal array access and unpacking
- Removed unnecessary packing of json util function in the SFN package
- Added related snapshot test

### CODE REVIEW

File: localstack-core/localstack/services/stepfunctions/asl/utils/json_path.py [hunk 1]
```diff
@@ -1,25 +1,37 @@
...
+    # Returns true if the json path terminates with a literal singleton array access.
+    return bool(re.search(_PATTERN_SINGLETON_ARRAY_ACCESS_OUTPUT, path))
 
-        if len(matches) > 1 or isinstance(matches[0].path, Index):
-            value = [match.value for match in matches]
-        else:
-            value = matches[0].value
 
-        return value
+def extract_json(path: str, data: json) -> json:
```
Comment (joe4dev) [inline localstack-core/localstack/services/stepfunctions/asl/utils/json_path.py:18]: pythonic 🐍 :)

File: tests/aws/services/stepfunctions/v2/base/test_base.py [hunk 2]
```diff
@@ -396,3 +396,30 @@ def test_state_pass_regex_json_path(
             definition,
             exec_input,
         )
+
+    @markers.aws.validated
+    @pytest.mark.parametrize(
+        "json_path_string",
+        ["$.items[0]", "$.items[10]"],
```
Comment (joe4dev) [inline tests/aws/services/stepfunctions/v2/base/test_base.py:403]: Would invalid or negative values throw an error?

File: localstack-core/localstack/services/stepfunctions/asl/utils/json_path.py [hunk 3]
```diff
@@ -1,25 +1,37 @@
 import json
+import re
+from typing import Final
 
 from jsonpath_ng.ext import parse
 from jsonpath_ng.jsonpath import Index
 
 from localstack.services.stepfunctions.asl.utils.encoding import to_json_str
 
+_PATTERN_SINGLETON_ARRAY_ACCESS_OUTPUT: Final[str] = r"\[\d+\]$"
```
Comment (joe4dev) [inline localstack-core/localstack/services/stepfunctions/asl/utils/json_path.py:10]: Did an error case reveal this regex or is that our best guess of how the AWS heuristic works?

**Review by joe4dev — APPROVED** [review]
Interesting special case. Well-done 🕵️

File: tests/aws/services/stepfunctions/v2/base/test_base.py (hunk 2 above)
Comment (MEPalma) [reply tests/aws/services/stepfunctions/v2/base/test_base.py:403]: I am adding a comment to remind us that the jsonpath library used for these operation in stepfunctions does not support negative or list indexing.

File: localstack-core/localstack/services/stepfunctions/asl/utils/json_path.py (hunk 3 above)
Comment (MEPalma) [reply localstack-core/localstack/services/stepfunctions/asl/utils/json_path.py:10]: The latter; this check appears to be restrictive enough for us to patch the output of the jsonpath library to match AWS's. However, this is indeed another sign that the jsonpath library does cover many of AWS SFN's logics

**Pull request merged**